### PR TITLE
Language code implementation

### PR DESCRIPTION
**Pull request by denisvieira-dev**

### CODE REVIEW

File: tests/Nox.Types.Tests/Types/LanguageCode/LanguageCodeTests.cs [hunk 1]
```diff
@@ -1,11 +1,66 @@
...
+
+        var languageCode2 = LanguageCode.From("EN");
+
+        languageCode2.Should().Be(languageCode1);
+    }
+
+
+    [Fact]
+    public void CreatingLanguageCode_CaseDifferentLanguageCode_ReturnsEqual()
+    {
```
Comment (andresharpe) [inline tests/Nox.Types.Tests/Types/LanguageCode/LanguageCodeTests.cs:40]: The class correctly allows case-insensitive creation of LanguageCode, but doesn't do a case insensitive comparison. I suggest we always internally convert language code Value to Invariant lower case, and change the Regex to be case sensitive.

File: src/Nox.Types/Types/LanguageCode/LanguageCode.cs [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
-    /// </summary>
-    /// <remarks>Placeholder, needs to be implemented</remarks>
-    // TODO Keep the List of Codes, find the proper ISO
-    public sealed class LanguageCode : ValueObject<string, LanguageCode>
+/// <summary>
+/// Class for two-letter language codes (ISO 639-1)
+/// </summary>
+public sealed class LanguageCode : ValueObject<string, LanguageCode>
+{
+    private static readonly Regex _languageCodeRegex = new(@"^([A][F]|SQ|AM|[A]R|[H]Y|AS|EU|[B]E|[B]N|[B]S|[B]R|[B]G|[M]Y|[K]M|[C]A|[C]H|[C]E|[N]Y|[Z]H|[C]O|[H]R|[C]S|[D]A|[N]L|[D]Z|[E]N|[E]T|[T]L|[F]I|[F]R|[F]Y|[G]L|[K]A|[D]E|[E]L|[G]U|[H]T|[H]A|[H]E|[H]I|[H]U|[I]S|[I]O|[G]A|[I]T|[J]A|[J]V|[K]N|[K]K|[R]W|[K]Y|[K]O|[K]U|[L]O|[L]V|[L]T|[L]B|[M]K|[M]G|[M]S|[M]L|[M]T|[M]I|[M]R|[M]N|[N]E|[N]O|[O]R|[P]S|[F]A|[P]L|[P]T|[P]A|[R]O|[R]U|[G]D|[S]R|[G]D|[S]I|[S]K|[S]L|[S]O|[S]O|[E]S|[S]W|[S]V|[T]A|[T]E|[T]H|[T]R|[U]K|[U]R|[V]I|[C]Y|[X]H|YI|ZU)$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
```
Comment (andresharpe) [inline src/Nox.Types/Types/LanguageCode/LanguageCode.cs:10]: Suggestion: Convert these to lowercase, then remove IgnoreCase from regex options. It also means that LanguageCode.From("EN") will return "en" in ToString() and Value and always persist as lowercase (the ISO standard representation)

File: src/Nox.Types/Types/LanguageCode/LanguageCode.cs [hunk 3]
```diff
@@ -1,10 +1,23 @@
...
+
+    internal override ValidationResult Validate()
     {
+        var result = base.Validate();
+
```
Comment (andresharpe) [inline src/Nox.Types/Types/LanguageCode/LanguageCode.cs:15]: Add: Value = Value.ToLowerInvariant();

**Review by andresharpe — CHANGES_REQUESTED** [review]

File: src/Nox.Types/Types/LanguageCode/LanguageCode.cs (hunk 3 above)
Comment (denisvieira-dev) [reply src/Nox.Types/Types/LanguageCode/LanguageCode.cs:15]: Hey @andresharpe, thanks for your suggestions. I updated the code based on them.

**Review by rochar — DISMISSED** [review]

File: src/Nox.Types/Types/LanguageCode/LanguageCode.cs [hunk 4]
```diff
@@ -1,10 +1,25 @@
...
-    /// </summary>
-    /// <remarks>Placeholder, needs to be implemented</remarks>
-    // TODO Keep the List of Codes, find the proper ISO
-    public sealed class LanguageCode : ValueObject<string, LanguageCode>
+/// <summary>
+/// Class for two-letter language codes (ISO 639-1)
+/// </summary>
+public sealed class LanguageCode : ValueObject<string, LanguageCode>
+{
+    private static readonly Regex _languageCodeRegex = new(@"^([a][f]|sq|am|[a]r|[h]y|as|eu|[b]e|[b]n|[b]s|[b]r|[b]g|[m]y|[k]m|[c]a|[c]h|[c]e|[n]y|[z]h|[c]o|[h]r|[c]s|[d]a|[n]l|[d]z|[e]n|[e]t|[t]l|[f]i|[f]r|[f]y|[g]l|[k]a|[d]e|[e]l|[g]u|[h]t|[h]a|[h]e|[h]i|[h]u|[i]s|[i]o|[g]a|[i]t|[j]a|[j]v|[k]n|[k]k|[r]w|[k]y|[k]o|[k]u|[l]o|[l]v|[l]t|[l]b|[m]k|[m]g|[m]s|[m]l|[m]t|[m]i|[m]r|[m]n|[n]e|[n]o|[o]r|[p]s|[f]a|[p]l|[p]t|[p]a|[r]o|[r]u|[g]d|[s]r|[g]d|[s]i|[s]k|[s]l|[s]o|[s]o|[e]s|[s]w|[s]v|[t]a|[t]e|[t]h|[t]r|[u]k|[u]r|[v]i|[c]y|[x]h|yi|zu)$", RegexOptions.Compiled);
```
Comment (gjorgji-grgovski-authoritypartners) [inline src/Nox.Types/Types/LanguageCode/LanguageCode.cs:10]: Can we change the Regex to HashSet?

**Review by gjorgji-grgovski-authoritypartners — CHANGES_REQUESTED** [review]

File: src/Nox.Types/Types/LanguageCode/LanguageCode.cs (hunk 4 above)
Comment (denisvieira-dev) [reply src/Nox.Types/Types/LanguageCode/LanguageCode.cs]: Done

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]

**Comment by gjorgji-grgovski-authoritypartners** [comment]
@rochar can you please review it again and complete the PR if all is good?

**Review by rochar — APPROVED** [review]
looks good

**Pull request merged**